src/Microsoft.Health.Dicom.Api/Features/Converter/CustomJsonStringEnumConverterImp.cs

```diff
@@ -0,0 +1,40 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Globalization;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using EnsureThat;
+
+namespace Microsoft.Health.Dicom.Api.Features.Converter
+{
+    internal class CustomJsonStringEnumConverterImp<T> : JsonConverter<T>
```

Comment:
nit: perhaps names like `EnumNameJsonConverter` for the factory and `EnumNameConverter` for the actual converter?